lib/python/qmk/cli/doctor.py

```diff
@@ -182,18 +195,17 @@ def check_udev_rules():
                     current_rules.add(line)
 
         # Check if the desired rules are among the currently present rules
-        for bootloader, rules in desired_rules.items():
+        for rules in desired_rules.items():
             # For caterina, check if ModemManager is running
-            if bootloader == "caterina":
-                if check_modem_manager():
-                    ok = False
-                    cli.log.warn("{bg_yellow}Detected ModemManager without the necessary udev rules. Please either disable it or set the appropriate udev rules if you are using a Pro Micro.")
+            if check_modem_manager():
```

Comment:
I'd rather keep the old behaviour. It doesn't make much sense to run the MM check for every rule when it only makes sense for the Caterina ones.

Reply:
I'll be honest, I don't really know what I'm doing with this section. I have to actually test it. I was trying to keep as much of the existing functionality as possible while still accounting for the fact `desired_rules` no longer has a `bootloader` part.

Reply:
Hm, doctor does not have the ability to write out `desired_rules` to a file, right? In that case keeping the bootloader dicts would make more sense.